### ISSUE TITLE
DevService for LRA

### ISSUE THREAD

**Issue by lordofthejars** [quarkusio/quarkus#23558]
### Description

When developing with MicroPofile LRA you need a coordinator. I think it is a candidate for DevServices so the coordinator is automatically started in Dev mode.

### Implementation ideas

Typical DevService starting LRA coordinator from Narayana project.

**Comment by geoand** [comment]
cc @mmusgrov

**Comment by mmusgrov** [comment]
Thanks @geoand

**Comment by lordofthejars** [comment]
Let me comment something about this issue:

To use an LRA coordinator, you cannot use TestContainers (un general Docker) as any service should e able to access to LRA coordinator (no problem when Docker) but LRA coordinator also needs to init a communication to Service, and here there is the problem. When you are at Docker you can access services inside Docker space, but from Docker space, you cannot access services running in the local machine (well let's say easily). So DevServices for LRA should run a local process.


About the process to run, Narayana is provided as a full zip with everything packed in but for the LRA coordinator you only need the few jars located at `narayana-full-5.12.4.Final/rts/lra`

**Comment by mmusgrov** [comment]
@lordofthejars you are correct, the communication between a service and an LRA coordinator must be bi-directional and `TestContainers` allows me to achieve that goal (see integration-tests/narayana-lra/src/test/java/org/acme/quickstart/lra/LRAParticipantTestResourceLifecycle.java):

``` java
To achieve that I used GenericContainer<?> registry = new GenericContainer<>("jbosstm/lra-coordinator:5.12.0.Final")
                .withExposedPorts(8080)
                .withEnv("QUARKUS_PROFILE", "prod");
```

**Comment by mmusgrov** [comment]
BTW, I have not changed anything, that was the original code which was merged back in Jul 16 2020

**Comment by lordofthejars** [comment]
So there is already a DevService for LRA? But no docs

**Comment by mmusgrov** [comment]
I'm not familiar with DevService's. I wrote the integration test to use TestContainers.
At the time I wrote the LRA extension nothing was available so I raised issue https://github.com/quarkusio/quarkus/issues/18425 which is still in the `Open` state.

**Comment by mmusgrov** [comment]
@geoand I'd like to clarify my understanding of what the intention of labelling an issue with `area/narayana` means. Does it mean that we need to fix the issue or that our input is required? If it is the former then we'd need some help and if it is the latter then I provided my input in the last message.

**Comment by geoand** [comment]
It's the latter but could lead to the former 😎.

If you think the issue has been addressed

**Comment by xstefank** [comment]
I will look into this.
